tests/test_jupyterhub_spawners.py

```diff
@@ -0,0 +1,37 @@
+from jupyterhub.app import JupyterHub
+from jupyterhub.spawner import Spawner
+from jupyterhub.tests.mocking import MockSpawner
+from traitlets.config import Config
+
+
+async def test_default_mockhub_instance_creator(get_configured_mockhub_instance):
+    hub_instance = get_configured_mockhub_instance()
+    assert isinstance(hub_instance, JupyterHub)
+    assert hub_instance.spawner_class == MockSpawner
+    assert hub_instance.config == {}
+
+
+async def test_default_hub_app(hub_app):
+    assert isinstance(hub_app, JupyterHub)
+    assert hub_app.spawner_class == MockSpawner
+
+
+async def test_mockhub_instance_creator(get_configured_mockhub_instance):
+    config_dict = {
+        "JupyterHub": {"spawner_class": Spawner},
+        "Spawner": {"home_dir_template": "/tmp/Jeff"},
+    }
+    new_hub_instance = get_configured_mockhub_instance(config=Config(config_dict))
+    assert isinstance(new_hub_instance, JupyterHub)
+    assert new_hub_instance.spawner_class == config_dict["JupyterHub"]["spawner_class"]
+    assert new_hub_instance.config == config_dict
+
+
+async def test_hub_app(hub_app):
+    assert isinstance(hub_app, JupyterHub)
+    config_dict = {
+        "JupyterHub": {"spawner_class": Spawner},
+        "Spawner": {"home_dir_template": "/tmp/Jeff"},
+    }
+    assert hub_app.spawner_class == config_dict["JupyterHub"]["spawner_class"]
```

Comment:
I am wondering why we need to define `config_dict` again in this test if the `hub_app` refers to the global `JupyterHub` instance.

Reply:
Also, is it possible to assert directly that the `spawner_class` has now been changed to `Spawner`?

Reply:
Like this:
```python
async def test_hub_app(hub_app):
    assert isinstance(hub_app, JupyterHub)
    assert hub_app.spawner_class == Spawner
    assert hub_app.config == {
        "JupyterHub": {"spawner_class": Spawner},
        "Spawner": {"home_dir_template": "/tmp/Jeff"},
    }
```

Reply:
Yes, sure. I chose to reference the `config_dict` instead because I thought that makes it more clear where the asserts come from and that is the same config like in the previous test.

Do you find it more useful and more clear to reference the config directly, and not store it in a variable?